studio/components/interfaces/Settings/Logs/LogTable.tsx

```diff
@@ -9,33 +9,43 @@ import { isNil } from 'lodash'
 
 interface Props {
   isCustomQuery: boolean
-  data?: LogData[]
+  data?: Array<LogData | Object>
 }
+type LogMap = { [id: string]: LogData }
 
 /**
  * Logs table view with focus side panel
+ *
+ * When in custom data display mode, the side panel will not open when focusing on logs.
  */
-const LogTable = ({ isCustomQuery, data }: Props) => {
+const LogTable = ({ isCustomQuery, data = [] }: Props) => {
   const [focusedLog, setFocusedLog] = useState<LogData | null>(null)
-  const columns = ['timestamp', 'event_message'].map((v) => ({
+  const columnNames = Object.keys(data[0] || {})
+  // whether it the data structure is LogData format.
+  const isChrono =
+    columnNames.includes('timestamp') &&
+    columnNames.includes('event_message') &&
+    columnNames.length === 4
+
+  const columns = (isChrono ? ['timestamp', 'event_message'] : columnNames).map((v) => ({
```

Comment:
I had to jump back to figure out what `isChrono` is

Perhaps something more descriptive like `hasLogDataFormat` or similar?

Reply:
changed, yes much more intuitive 😅